autogpt_platform/backend/backend/blocks/github/checks.py

```diff
@@ -0,0 +1,354 @@
+            description="Title of the check run output",
+            default=None,
+        )
+        output_summary: Optional[str] = SchemaField(
+            description="Summary of the check run output",
+            default=None,
+        )
+        output_text: Optional[str] = SchemaField(
+            description="Detailed text of the check run output",
+            default=None,
+        )
+
+    class Output(BlockSchema):
+        class CheckRunResult(TypedDict):
+            id: int
+            html_url: str
+            status: str
+
+        check_run: CheckRunResult = SchemaField(
+            description="Details of the created check run"
+        )
+        error: str = SchemaField(
+            description="Error message if check run creation failed"
+        )
+
+    def __init__(self):
+        super().__init__(
+            id="2f45e89a-3b7d-4f22-b89e-6c4f5c7e1234",  # Generated UUID
+            description="Creates a new check run for a specific commit in a GitHub repository",
+            categories={BlockCategory.DEVELOPER_TOOLS},
+            input_schema=GithubCreateCheckRunBlock.Input,
+            output_schema=GithubCreateCheckRunBlock.Output,
+            test_input={
+                "repo_url": "https://github.com/owner/repo",
+                "name": "test-check",
+                "head_sha": "ce587453ced02b1526dfb4cb910479d431683101",
+                "status": ChecksStatus.COMPLETED.value,
+                "conclusion": ChecksConclusion.SUCCESS.value,
+                "output_title": "Test Results",
+                "output_summary": "All tests passed",
+                "credentials": TEST_CREDENTIALS_INPUT,
+            },
+            # requires a github app not available to oauth in our current system
+            # disabled=True,
+            test_credentials=TEST_CREDENTIALS,
+            test_output=[
+                (
+                    "check_run",
+                    {
+                        "id": 4,
+                        "html_url": "https://github.com/owner/repo/runs/4",
+                        "status": "completed",
+                    },
+                ),
+            ],
+            test_mock={
+                "create_check_run": lambda *args, **kwargs: {
+                    "id": 4,
+                    "html_url": "https://github.com/owner/repo/runs/4",
+                    "status": "completed",
+                }
+            },
+        )
+
+    @staticmethod
+    def create_check_run(
+        credentials: GithubCredentials,
+        repo_url: str,
+        name: str,
+        head_sha: str,
+        status: ChecksStatus,
+        conclusion: Optional[ChecksConclusion] = None,
+        details_url: Optional[str] = None,
+        output_title: Optional[str] = None,
+        output_summary: Optional[str] = None,
+        output_text: Optional[str] = None,
+    ) -> dict:
+        api = get_api(credentials)
+
+        class CheckRunData(TypedDict, total=False):
```

Comment:
if `total=False`, iirc you have to use `Required[T]` to mark the actually required properties as such